### ISSUE TITLE
Prototype is Losted

### ISSUE THREAD

**Issue by xswei** [philipstanislaus/performant-array-to-tree#34]
https://github.com/philipstanislaus/performant-array-to-tree/blob/efe07a8694e74a7dd114149961e2baa4db0c4397/src/arrayToTree.ts#L88

change

```js
lookup[itemId] = {
    ...item,
    [conf.childrenField]: lookup[itemId][conf.childrenField],
  };
```

to:

```js
lookup[itemId] = Object.assign(item, {
  [conf.childrenField]: lookup[itemId][conf.childrenField]
})
```

**Comment by xswei** [comment]
When item is an object that inherits other classes, the method of the parent class is missing

**Comment by philipstanislaus** [comment]
Great Point, could you open a PR for this please?

**Comment by philipstanislaus** [comment]
Released in https://www.npmjs.com/package/performant-array-to-tree/v/1.11.0 , closed through #44
